pg_search/src/postgres/customscan/pdbscan/qual_inspect.rs

```diff
@@ -524,6 +543,19 @@ pub unsafe fn extract_quals(
             })
         }
 
+        pg_sys::NodeTag::T_NullTest => {
+            let nulltest = nodecast!(NullTest, T_NullTest, node)?;
+            if (*nulltest).nulltesttype == pg_sys::NullTestType::IS_NOT_NULL {
+                Some(Qual::PushdownIsNotNull {
+                    attname: attname_from_var(root, (*nulltest).arg.cast())
+                        .1
+                        .expect("var should have an attname"),
+                })
```

Comment:
What I might have done here is instead of also inventing Qual::PushdownIs**Not**Null, I probably would have wrapped this with the existing `Qual::Not()`.

I _think_ the boolean logic is the same between `field IS NOT NULL` and `NOT field IS NULL`.

Did you try this?  If not, I'd like to see this changed.

Reply:
hey @eeeebbbbrrrr thanks for the review.

I agree that we can represent `field IS NOT NULL` as `NOT field IS NULL` or `Qual::Not(field IS NULL)`. However, for this to work, we need a `Qual` variant that can handle predicates of type `field IS NULL`. Currently, my implementation only includes `Qual::PushdownIsNotNull` for `field IS NOT NULL`.

To confirm, are you expecting:
- A new `Qual` variant to handle `field IS NULL`, e.g., `Qual::PushdownIsNull`.
- Then, represent `field IS NOT NULL` as `Qual::Not( Qual::PushdownIsNull )` ?

Please let me know if this is the direction you’re expecting, and I’ll proceed with the changes accordingly.

Reply:
I should also note that Tantivy currently doesn't provide a direct way to check if a `field IS NULL` or has a `NotExists` query. It does have an `Exists` query (which I'm using for `IS NOT NULL`), but I'll need to determine the best approach for implementing the `IS NULL` check if we want to go with the `Qual::Not()` approach.